packages/messenger-widget/src/components/ConfigureProfile/bl.tsx

```diff
@@ -226,21 +220,28 @@ export const namingServices = [
     {
         name: NAME_SERVICES.ENS,
         chainId: 1,
-        component: <ConfigureEnsProfile />,
     },
     {
         name: NAME_SERVICES.GENOME,
         chainId: 100,
-        component: <ConfigureGenomeProfile />,
     },
 ];
 
 export const fetchComponent = (name: string) => {
+    let chain: number = 1;
     switch (name) {
         case NAME_SERVICES.ENS:
-            return namingServices[0].component;
+            if (process.env.REACT_APP_CHAIN_ID === '5') {
+                chain = Number(process.env.REACT_APP_CHAIN_ID);
+            } else {
+                chain = namingServices[0].chainId;
+            }
+            return <ConfigureEnsProfile chainToConnect={chain} />;
         case NAME_SERVICES.GENOME:
-            return namingServices[1].component;
+            chain = namingServices[1].chainId;
+            return <ConfigureGenomeProfile chainToConnect={chain} />;
+        default:
```

Comment:
This will always fails since there is no chain 1

Reply:
```
export const namingServices = [
    {
        name: NAME_SERVICES.ENS,
        chainId: 1,
    },
    {
        name: NAME_SERVICES.GENOME,
        chainId: 100,
    },
];

export const fetchComponent = (name: string) => {
    let chain: number = 1;
    switch (name) {
        case NAME_SERVICES.ENS:
            if (process.env.REACT_APP_CHAIN_ID === '5') {
                chain = Number(process.env.REACT_APP_CHAIN_ID);
            } else {
                chain = namingServices[0].chainId;
            }
            return <ConfigureEnsProfile chainToConnect={chain} />;
        case NAME_SERVICES.GENOME:
            chain = namingServices[1].chainId;
            return <ConfigureGenomeProfile chainToConnect={chain} />;
        default:
            return <ConfigureEnsProfile chainToConnect={chain} />;
    }
};```

Here as there are two chains, Ethereum mainnet with chain ID 1 and Gnosis with chain id 100. If user is connected to goerli for development it returns ENS that is  namingServices[0] with chain ID 5. If connected to mainnet then ENS with chainID 1 and if Gnosis then returns with chainID 100.

What's wrong here?

Reply:
Here as there are two chains, Ethereum mainnet with chain ID 1 and Gnosis with chain id 100. If user is connected to goerli for development it returns ENS that is  namingServices[0] with chain ID 5. If connected to mainnet then ENS with chainID 1 and if Gnosis then returns with chainID 100.

What's wrong here?

Reply:
The default case will be never reached because it'll always return earlier.  Is case the chain is mainnet it'll return at the end of the if statement. The additional return and the let var are adding unneccessary complexity to the function and makes it harder to understand it